gtk4/src/closure_expression.rs

```diff
@@ -23,18 +23,22 @@ define_expression!(
 
 impl ClosureExpression {
     #[doc(alias = "gtk_closure_expression_new")]
-    pub fn new<F, R>(params: impl IntoIterator<Item = Expression>, callback: F) -> Self
+    pub fn new<F, R, I>(params: impl IntoIterator<Item = I>, callback: F) -> Self
     where
         F: Fn(&[Value]) -> R + 'static,
         R: ValueType,
+        I: AsRef<Expression> + Clone,
```

Comment:
Why not `impl IntoIterator<Item = impl AsRef<Expression>>`?

Reply:
I think it's a bit more readable, but maybe just me

Reply:
Something for @bilelmoussaoui to decide then. Looks good to me apart from this :)

Reply:
Why does it need `Clone` though? `&Expression` is `Clone`

Reply:
The compiler used to complain about it a while ago, but somehow it's now okay

Reply:
> Why not `impl IntoIterator<Item = impl AsRef<Expression>>`

I'm also in favour of using something like this btw